src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -849,6 +880,7 @@ def parent_construct_payload(self):
         self.set_up_registry_identity()
 
     def construct_payload(self):
+        self.set_up_system_assigned_identity_as_default_if_using_acr()
```

Comment:
why not put after parent_construct_payload()?

Reply:
The `set_up_system_assigned_identity_as_default_if_using_acr` will set --identity to `system` as default, it should be execute before parent_construct_payload()